.github/workflows/deploy.yml

```diff
@@ -26,67 +11,74 @@ on:
     inputs:
       BRANCH:
         description: "Branch/tag to deploy"
-        options:
-          - stage
-          - test
-          - master
         default: stage
         required: true
       environment:
-        description: "target environment to deploy to"
+        description: "Target environment to deploy to"
         type: choice
         options:
           - staging
           - agroportal
           - test
-        default: stage
+        default: staging
+
 jobs:
   deploy:
     runs-on: ubuntu-latest
     env:
-      BUNDLE_WITHOUT: default #install gems required primarely for deployment in order to speed up workflow
+      BUNDLE_WITHOUT: default # Install gems required primarily for deployment to speed up workflow
       PRIVATE_CONFIG_REPO: ${{ format('git@github.com:{0}.git', secrets.CONFIG_REPO) }}
+
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      - name: set branch/tag and environment to deploy from inputs
+      - name: Set branch/tag and environment to deploy from inputs
         run: |
-          # workflow_dispatch default input doesn't get set on push so we need to set defaults
-          # via shell parameter expansion
-          # https://dev.to/mrmike/github-action-handling-input-default-value-5f2g
-          USER_INPUT_BRANCH=${{ inputs.branch }}
-          echo "BRANCH=${USER_INPUT_BRANCH:github.head_ref:-master}" >> $GITHUB_ENV
+          # Set default branch/tag for push events
+          BRANCH=${{ github.head_ref || 'master' }}
+          echo "BRANCH=$BRANCH" >> $GITHUB_ENV
 
-          USER_INPUT_ENVIRONMENT=${{ inputs.environment }}
-          echo "TARGET=${USER_INPUT_ENVIRONMENT:-staging}" >> $GITHUB_ENV
+          # Set default environment for push events
+          if [ "${{ github.event_name }}" = "push" ]; then
+            TARGET="staging" # Default to staging for push
```

Comment:
don't default push into stage, it need to check the current pushed branch and depending on it decide where to push